plugins/source/aws/codegen/recipes/elasticsearch.go

```diff
@@ -7,26 +7,65 @@ import (
 )
 
 func ElasticsearchResources() []*Resource {
-	resources := []*Resource{
+	return []*Resource{
 		{
-			SubService:          "domains",
-			Struct:              &types.ElasticsearchDomainStatus{},
-			PreResourceResolver: "getDomain",
-			PKColumns:           []string{"arn"},
-			ExtraColumns: append(defaultRegionalColumns,
+			Service:     "elasticsearch",
+			SubService:  "domains",
+			Struct:      new(types.ElasticsearchDomainStatus),
+			Description: "https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_DomainStatus.html",
+			Multiplex:   `client.ServiceAccountRegionMultiplexer("es")`,
+			PKColumns:   []string{"arn"},
+			ExtraColumns: append(
+				defaultRegionalColumns,
+				codegen.ColumnDefinition{
+					Name:     "authorized_principals",
+					Type:     schema.TypeJSON,
+					Resolver: `resolveAuthorizedPrincipals`,
+				},
 				codegen.ColumnDefinition{
 					Name:     "tags",
 					Type:     schema.TypeJSON,
-					Resolver: `resolveTags`,
+					Resolver: `resolveDomainTags`,
 				},
 			),
 		},
+		{
+			Service:      "elasticsearch",
+			SubService:   "packages",
+			Struct:       new(types.PackageDetails),
+			Description:  "https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_PackageDetails.html",
+			Multiplex:    `client.ServiceAccountRegionMultiplexer("es")`,
+			PKColumns:    []string{"package_name"},
+			ExtraColumns: defaultRegionalColumnsPK,
+		},
+		{
+			Service:     "elasticsearch",
+			SubService:  "versions",
+			Struct:      new(struct{}),
+			Description: "https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_ListVersions.html",
+			Multiplex:   `client.ServiceAccountRegionMultiplexer("es")`,
+			PKColumns:   []string{"version"},
+			ExtraColumns: append(defaultRegionalColumnsPK,
+				codegen.ColumnDefinition{
+					Name:     "version",
+					Type:     schema.TypeString,
+					Resolver: `resolveVersion`,
+				},
+				codegen.ColumnDefinition{
+					Name:     "instance_types",
+					Type:     schema.TypeJSON,
+					Resolver: `resolveInstanceTypes`,
+				},
+			),
+		},
+		{
+			Service:      "elasticsearch",
+			SubService:   "vpc_endpoints",
+			Struct:       new(types.VpcEndpoint),
+			Description:  "https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_VpcEndpoint.html",
+			Multiplex:    `client.ServiceAccountRegionMultiplexer("es")`,
+			PKColumns:    []string{"vpc_endpoint_id"},
```

Comment:
In other cases like this one in AWS, we opted to rename the field to `id`. See for example [here](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/aws/codegen/recipes/apigatewayv2.go#L24-L28) or [here](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/aws/codegen/recipes/directconnect.go#L25-L30). Let's keep doing that for consistency.

The rule of thumb I follow is basically: keep column names the same as they are on the original struct, unless they are something like `<StructName>ID` or `<StructName>ARN`, in which case it should be renamed to `id` or `arn`--especially if it's part of the PK, since then joins will often need to be performed using these so it's better if it's easily guessable.

Reply:
changed to `id`, also updated pk for `packages`